### DISCUSSION TITLE
Move connection string in config settings

### DISCUSSION THREAD

**Discussion by rudolfdeschipper** [quartznet/quartznet#2535] [Q&A] [answered]
I need to get the database connection string from the normal application settings, as a regular connection string (this is required for security reasons, so I can store the connection string in an Azure App instance).
All examples for Quarts show the connection string  literally in the settings or the config.

Since v3.13, the config seems not to be read anymore from the applications settings - Quartz reports that the datasource is unknown or cannot be found.

I noted that the setting ConnectionStringName, which is supposed to retrieve the connection string from the application settings, always returns NULL, which probably is the reason that connection can't be set up anymore. It is not clear why this was done, and what the new way of working should be.

How to overcome this?

Thanks!

**Comment by lahma** [comment]
Can you elaborate a bit more. 

* Target framework: NET 8?
* Location of connection string. `MyApp.exe.config` or similar? Connection strings section or `appSettings`?

This changed in 3.13 as you can see in [release notes](https://github.com/quartznet/quartznet/releases/tag/v3.13.0). Because I was aware that someone would want to mix modern runtime and old configuration, you can always derive from `StdSchedulerFactory` and override `protected virtual string? GetNamedConnectionString(string connectionStringName)`.

**Reply by rudolfdeschipper** [reply]
Thanks for the quick response!

I'm on .NET 8, Core
Connection string is located in the appsetting.json of a backend API service.
As the whole is running on Azure, the connection strings are externalized (Azure takes care of making them available to the environemnt configuration) 
I was not aware there is an "old" and "modern" approach. I'm using the quartz.config settings mostly, but the db connection string must come out of the config. I can't have it in any of the code files (which will end up in github, thus giving anyone access to the connection string).

Hope this helps.

**Reply by lahma** [reply, marked as answer]
I have to do a lot of guesswork here as there's no concrete examples of what you are doing. Are you using the DI integration or using StdSchedulerFactory directly? DI integration has named connection string resolving path from configuration.

**Reply by rudolfdeschipper** [reply]
I'm using the StdSchedulerFactory - I'll fiddle around with DI, sounds like a better option anyway. Thanks!

**Comment by rudolfdeschipper** [comment]
Closing for now. Will get back if further issues.
